core/croquet/src/main/java/org/lgna/croquet/CascadeBlank.java

```diff
@@ -84,16 +85,10 @@ public final CascadeBlankChild[] getFilteredChildren(BlankNode<B> blankNode) {
       }
     }
 
-    //remove separators at the end
-    //there should be a maximum of only 1 but we loop anyway
-    final int N = children.size();
-    for (int i = 0; i < N; i++) {
-      int index = N - i - 1;
-      if (isEmptySeparator(children.get(index))) {
-        children.remove(index);
-      } else {
-        break;
-      }
+    //remove separators at the end. there shouldn't be more than one, since we just removed all the duplicates.
+    final int last = children.size() - 1;
+    if (isEmptySeparator(children.get(last))) {
```

Comment:
Much simpler, but it breaks when the children are empty.
To test it add a field to the scene and try to set the initializer before setting the type. You will see the red queen.

Reply:
oof, thanks.  i think i used all the braincells untangling the previous logic and didn't recover when writing mine.  I'll add a check